libs/response-lib.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Handle a successfuly API call return case.
+ */
+export function success(statusCode, body) {
+  return buildResponse(statusCode, body);
+}
+
+/**
+ * Handle a failed API call return case.
+ */
+export function failure(statusCode, body) {
```

Comment:
Imagine a scenario where the client requires each HTTP failure status code to have a custom `customMessage` property. The value of this property must be retrieved from a standalone API.

How could that mechanism be incorporated to `response-lib`?

Reply:
As this code is laid out, its very basic and strict with respect to the code/body parameters.  I'd hate to have a function that required a large amount of parameters, but I'd also be wary of taking in an object where all the properties need to be inspected and then matched to the response.  There is likely a happy medium here, adding in a customMessage parameter doesn't seem like to much scope creep for this function but it could quickly lead to some bad code smells that becomes difficult to maintain if others followed suit.  A custom message could also have a null default value that wouldn't require it be provided to keep it clean when others don't have this requirement.